src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4956,8 +4956,8 @@ Pipe MergeTreeData::alterPartition(
                 if (command.replace)
                     checkPartitionCanBeDropped(command.partition, query_context);
 
-                String from_database = query_context->resolveDatabase(command.from_database);
-                auto from_storage = DatabaseCatalog::instance().getTable({from_database, command.from_table}, query_context);
+                auto resolved = query_context->resolveStorageID({command.from_database, command.from_table});
+                auto from_storage = DatabaseCatalog::instance().getTable(resolved, query_context);
```

Comment:
Okay, but it will not work with `ON CLUSTER` queries and `Replicated` databases. Also it will be less optimal with `ReplicatedMergeTree` because it will be unable to find the table on other replicas